pyo3-build-config/src/errors.rs

```diff
@@ -12,15 +12,21 @@ macro_rules! ensure {
     ($condition:expr, $($args: tt)+) => { if !($condition) { bail!($($args)+) } };
 }
 
-/// Show warning. If needed, please extend this macro to support arguments.
+/// Show warning.
 #[macro_export]
 #[doc(hidden)]
 macro_rules! warn {
-    ($msg: literal) => {
-        println!(concat!("cargo:warning=", $msg))
+    ($($args: tt)+) => {
+        println!("{}", $crate::format_warn!($($args)+))
     };
-    ($fmt: expr, $($args: tt)+) => {
-        println!("cargo:warning={}", format_args!($fmt, $($args)+))
+}
+
+/// Show warning on a specific writer.
+#[macro_export]
+#[doc(hidden)]
+macro_rules! format_warn {
+    ($($args: tt)+) => {
+        format!("cargo:warning={}", format_args!($($args)+))
```

Comment:
The doc comment does not seem to apply any more as this yields a `String`. Also isn't the `format_args!` layer unnecessary again?

Reply:
Ah sorry, the `format_args!` is necessary because this includes another format string at the call site. I force pushed a fix to the comment.